lib/mallet.js

```diff
@@ -124,6 +125,120 @@ class Iele {
     tx.data = '0x' + rlp.encode([code, args]).toString('hex');
     return this.mallet.sendTransaction(tx, password);
   }
+
+  compileSol(source) {
+    const request = new XMLHttpRequest();
+    request.open('POST', testnets.iele.compiler, false);
+    request.setRequestHeader("Content-Type", "application/json");
+    request.responseType = 'json';
+
+    const payload = {
+      "jsonrpc": "2.0", 
+      "method": "sol2iele_asm",
+      "id": 0,
+      "params": [
+        "main.sol", 
+        {
+          "main.sol": source
+        }
+      ]
+    }
+    request.send(JSON.stringify(payload));
+
+    let output = null;
+    if (request.status == 200) {
+      let responseJson = null;
+      try {
+        responseJson = JSON.parse(request.responseText);
+      } catch (err) {
+        throw 'Compiler error: ' + err;
+      }
+
+      if (responseJson.error) {
+        throw 'Compiler error: ' + JSON.stringify(responseJson.error);
+      } else {
+        output = responseJson.result;
+      }
+    } else {
+      throw 'Compiler error: ' + request.status + '\n' + request.responseText;
+    }
+
+    let codeStarted = false;
+    let code = '';
+
+    for (let line of output.split('\n')) {
+      if (codeStarted) {
+        code += line + '\n'
+      } else {
+        codeStarted = line === 'IELE assembly:'
+      }
+    }
+
+    const result = {source: source, solidityCompilerOutput: output, error: false}
+
+    if (codeStarted) {
+      result.ieleCode = code;
+      const bytecode = this.compileIele(code).result;
+      result.bytecode = bytecode;
+      return result;
+    } else {
+      result.error = true;
+      return result;
+    }
+  }
+
+  compileIele(source) {
```

Comment:
this method is quite similar in structure to compileSol. Can we extract & reuse some code?